subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -1569,6 +1570,10 @@ int lwm2m_rd_client_start(struct lwm2m_ctx *client_ctx, const char *ep_name,
 	client.ctx->sec_obj_inst = -1;
 	client.retries = 0;
 
+#if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT)
+	lwm2m_pull_context_set_sockopt_callback(client.ctx->set_socketoptions);
```

Comment:
While I'm fine with most of the changes in this PR, I don't think it's actually ok to set the same `set_socketoptions` callback for the firmware ctx as it is for the regular LwM2M client context? One of the purpose of this callback is to configure custom TLS credentials/options for the context and I imagine those could be different for the LwM2M server and firmware download server?

Reply:
It doesn't look like this callback is used anywhere in tree.
TLS credentials for the LwM2M client would be configured through the security object. The LwM2M library takes care of configuring the socket with these credentials.

I can see there being additional options that could be different. I can remove this call and instead add this to the public LwM2M API. Would you be okay with that?

Reply:
Not in tree, but we use it in nRF Connect SDK for instance.

> I can see there being additional options that could be different. I can remove this call and instead add this to the public LwM2M API. Would you be okay with that?

Sounds fine to me.